locustempus/main/serializers.py

```diff
@@ -34,12 +35,49 @@ class ProjectSerializer(serializers.ModelSerializer):
         many=True
     )
 
+    aggregated_layers = serializers.SerializerMethodField()
+
+    def get_aggregated_layers(self, obj):
+        # Returns project layers PLUS
+        # - If the current user is faculty, it adds nothing, just returns
+        #   the project layers
+        # - If the current user is a student AND there's an Activity, it
+        #   concats layers related to other student's response, excluding
+        #   the current user, but only if the other student's response is
+        #   submitted or reviewed
+        request = self.context['request']
+        user = request.user
+        course = obj.course
+
+        if course.is_true_faculty(user):
+            return [reverse('api-layer-detail', args=[lyr.pk], request=request)
+                    for lyr in obj.layers.all()]
+
+        elif hasattr(obj, 'activity') and course.is_true_member(user):
+            all_layers = [
```

Comment:
In the rendered Base Layers view, ideally, the instructor's layers would sort first, then the student layers sorted by last, first / username. I'm not sure if you want to tackle the sorting up here on the server or down at the client.

Reply:
I agree, the Author's layers should be on to at all times, contributors should be of some order as well